### PR TITLE
HPCC-26995 Minor speed improvement processing #directives

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@shamser this is a purely mechanical change - avoiding creating a string until needed.  See jira for more details.

**Review by richardkchapman — APPROVED** [review]

File: ecl/hql/hqlparse.cpp [hunk 1]
```diff
@@ -650,7 +650,7 @@ bool HqlLex::getParameter(StringBuffer &curParam, const char* for_what)
         case EOF:
             {
                 StringBuffer msg("EOF encountered while gathering parameters for ");
-                msg.append(for_what);
+                msg.appendf("%s(%u,%u)",directive, location.lineno, location.column);
```
Comment (shamser) [inline ecl/hql/hqlparse.cpp:653]: I understand reportError is printing the line/column from nextToken. I am not sure that the location needs to be appended to msg.  I mean this for 653-654:
msg.appendf("%s",directive);
reportError(**location**, ERR_TMPLT_EOFINPARAM, "%s", msg.str());

File: ecl/hql/hqlparse.cpp [hunk 2]
```diff
@@ -1205,7 +1201,7 @@ void HqlLex::doFor(attribute & returnToken, bool doAll)
 
         if (tok == EOF)
         {
-            reportError(returnToken, ERR_TMPLT_HASHENDEXPECTED, "EOF encountered inside %s: #END expected", forwhat.str());
+            reportError(returnToken, ERR_TMPLT_HASHENDEXPECTED, "EOF encountered inside %s(%u,%u): #END expected", directive, location.lineno, location.column);
```
Comment (shamser) [inline ecl/hql/hqlparse.cpp:1204]: Same as comment on line 643.

File: ecl/hql/hqlparse.cpp [hunk 3]
```diff
@@ -1670,7 +1669,7 @@ bool HqlLex::getDefinedParameter(StringBuffer &curParam, attribute & returnToken
         case EOF:
             {
                 StringBuffer msg("EOF encountered while gathering parameters for ");
-                msg.append(for_what);
+                msg.appendf("%s(%u,%u)",directive, location.lineno, location.column);
```
Comment (shamser) [inline ecl/hql/hqlparse.cpp:1672]: Same as comment on line 653.

**Review by shamser — CHANGES_REQUESTED** [review]
@ghalliday I believe line/column is being duplicated in the message as reportError is also output the location.

File: ecl/hql/hqlparse.cpp (hunk 1 above)
Comment (ghalliday) [reply ecl/hql/hqlparse.cpp]: I agree that would be better - I will change.

File: ecl/hql/hqlparse.cpp (hunk 3 above)
Comment (shamser) [reply ecl/hql/hqlparse.cpp:1664]: This one todo still.

**Review by shamser — CHANGES_REQUESTED** [review]
There is one comment that hasn't been addressed.  Also, it's failing to build for some reason.

**Review by shamser — APPROVED** [review]
Reviewed.  Looks good.

**Pull request merged**